definitions/procedures/packages/update.rb

```diff
@@ -11,7 +11,7 @@ class Update < ForemanMaintain::Procedure
 
     def run
       assumeyes_val = @assumeyes.nil? ? assumeyes? : @assumeyes
-      package_manager.clean_cache(:assumeyes => assumeyes_val)
+      package_manager.clean_cache(:assumeyes => assumeyes_val) if download_only?
```

Comment:
IMHO we should rather fix the definition of `clean_cache` in https://github.com/theforeman/foreman_maintain/blob/fa2e3cb52fc6d224124bd9d3e9d1c888578625a8/lib/foreman_maintain/package_manager/dnf.rb#L94-L96

`clean metadata` should be usually sufficient as it cleans everything but packages: https://github.com/rpm-software-management/dnf/blob/b5d25c075ba12d7543a4fb6ac39c6b20e5e5088a/dnf/cli/commands/clean.py#L40-L46

Reply:
Or maybe both. Re-reading your explanation above makes sense not to clean the cache "inbetween", but even for the non-inbetween case we should not `clean all` IMHO